lib/Doctrine/DBAL/Types/ObjectType.php

```diff
@@ -58,9 +60,15 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
         }
 
         $value = (is_resource($value)) ? stream_get_contents($value) : $value;
-        $val = unserialize($value);
-        if ($val === false && $value !== 'b:0;') {
-            throw ConversionException::conversionFailed($value, $this->getName());
+
+        set_error_handler(function (int $errno, string $errstr) use ($value) {
```

Comment:
The last nitpick, please add `: void` to the fn signature and squash all commits. The test looks fine.

Reply:
Also `$code` instead of `$errno` and `$message` instead of `$errstr` would be more readable. :yum: